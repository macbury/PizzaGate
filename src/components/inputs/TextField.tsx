import React, { useState } from 'react'
import { TextInputProps } from 'react-native'
import { useTranslation } from 'react-i18next'
import styled from 'styled-components/native'

interface ITextInputContainerProps {
  focused: boolean
  editable?: boolean
  hasError?: boolean
  theme?: any
}

function getLabelColor({ focused, hasError, theme } : ITextInputContainerProps) {
  if (hasError) {
    return theme.colors.error
  } else if (focused) {
    return theme.colors.primary
  } else {
    return theme.colors.label
  }
}

function getBorderColor({ focused, hasError, theme } : ITextInputContainerProps) {
  if (hasError) {
    return theme.colors.error
  } else if (focused) {
    return theme.colors.primary
  } else {
    return theme.colors.inputBorder
  }
}

const TextInput = styled.TextInput`
  color: ${({ theme }) => theme.colors.inputText};
  padding-top: 2px;
  padding-left: 20px;
  padding-bottom: 8px;
  background: rgba(0,0,0,0);
  font-size: ${({ theme }) => theme.fontSize.inputText};
  font-family: ${({ theme }) => theme.font.main};
`

const Label = styled.Text`
  padding-top: 8px;
  font-family: ${({ theme }) => theme.font.main};
  color: ${getLabelColor};
  padding-left: 20px;
  padding-right: 20px;
  font-size: ${({ theme }) => theme.fontSize.label};
`

const ErrorMessage = styled.Text`
  padding-top: 5px;
  font-family: ${({ theme }) => theme.font.main};
  color: ${({ theme }) => theme.colors.error};
  font-size: ${({ theme }) => theme.fontSize.label};
  padding-left: 20px;
  padding-right: 20px;
`

const TextInputContainer = styled.View`
  opacity: ${({ editable }) => editable ? '1.0' : '0.6'};
  background-color: ${({ theme }) => theme.colors.inputBackground};
  border-color: ${getBorderColor};
  border-radius: 5px;
  border-bottom-width: 2px;
`

const Wrapper = styled.View`
  margin-bottom: 15px;
`

export interface ITextFieldProps extends TextInputProps {
  label?: string
  editable?: boolean
  error?: string
  className?: string
  placeholder?: string
  value: string
  hidden?: boolean
  secureTextEntry?: boolean
  theme?: any
  onChangeText(input : string)
}

export default function TextField(props : ITextFieldProps) {
  const { t } = useTranslation()
  const [isFocused, setFocused] = useState(false)

  const {
    label,
    value,
    editable,
    placeholder,
    secureTextEntry,
    onChangeText,
    hidden,
    error,
    className,
    ...rest
  } = props

  if (hidden) {
    return null
  }

  return (
    <Wrapper {...{ className }}>
      <TextInputContainer hasError={!!error} editable={editable} focused={isFocused}>
        <Label hasError={!!error} focused={isFocused}>{t(label)}</Label>
        <TextInput
          {...rest}
          onFocus={() => setFocused(true)}
          onBlur={() => setFocused(false)}
          editable={editable}
          placeholder={placeholder}
          secureTextEntry={secureTextEntry}
          value={value}
          onChangeText={onChangeText} />
      </TextInputContainer>
      {error && <ErrorMessage>{error}</ErrorMessage>}
    </Wrapper>
  )
}