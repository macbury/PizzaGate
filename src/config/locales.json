{
  "en": {
    "screens": {
      "calculator": {
        "title": "Design your pizza",
        "numberOfBalls": "Dough balls number",
        "piece": "pieces",
        "doughWeight": "Dough weight",
        "wantedHydration": "Hydration",
        "wantedFats": "Oil",
        "wantedSalt": "Salt",
        "temperature": "Room temperature",
        "risingTime": "Rising time",
        "fridgeRisingTime": "Fridge rising time"
      },
      "showRecipe": {
        "title": "Here is your recipe!",
        "flour": "Flour:",
        "water": "Water:",
        "salt": "Salt:",
        "oil": "Oil:",
        "yeast": {
          "title": "Yeast:",
          "fresh": "g(fresh)",
          "dry": "g(dry)"
        },
        "risingTime": {
          "title": "Rising Time:",
          "unit": "h"
        },
        "temperature": "Rising Temperature:",
        "fridgeRisingTime": {
          "title": "Fridge rising time",
          "unit": "h"
        }
      }
    }
  }
}