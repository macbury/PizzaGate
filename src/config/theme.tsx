import React, { useMemo } from 'react'
import { StatusBar } from 'react-native'
import { useMediaQuery } from 'react-responsive'
import { ThemeProvider } from 'styled-components/native'
import useSetNavBarColor from '../helpers/useSetNavBarColor'
import GlobalStyles from '../components/GlobalStyles'
import useSystemTheme from '../helpers/useSystemTheme'
type Device = 'desktop' | 'mobile'

export type Theme = {
  device: Device;
  dark: boolean;
  font: {
    main: string,
  },
  fontSize: {
    searchInput: string;
    text: string;
    label: string;
    inputText: string;
    buttonText: string;
  },
  opacity: {
    buttonClick: number;
  },
  colors: {
    cardBackground: string,
    tabActive: string,
    placeholder: string;
    navigationBar: string;
    statusBar: string;
    notificationTextColor: string;
    error: string;
    label: string;
    buttonTextColor: string;
    inputText: string;
    inputBackground: string;
    inputBorder: string;
    primary: string;
    background: string;
    card: string;
    text: string;
    border: string;
    notification: string;
  };
}

export const light : Theme = {
  device: 'desktop',
  dark: false,
  font: {
    main: 'Roboto-Regular'
  },
  fontSize: {
    text: '16px',
    label: '15px',
    searchInput: '16px',
    inputText: '19px',
    buttonText: '15px'
  },
  opacity: {
    buttonClick: 0.6
  },
  colors: {
    tabActive: '#fff',
    placeholder: 'rgba(0, 0, 0, 0.3)',
    statusBar: '#b44756',
    navigationBar: '#000000',
    notificationTextColor: '#fff',
    error: 'rgb(224, 36, 94)',
    buttonTextColor: '#fff',
    inputText: '#333',
    inputBackground: 'rgba(0, 0, 0, 0.05)',
    inputBorder: 'rgba(0, 0, 0, 0.08)',
    label: 'rgba(0, 0, 0, 0.5)',
    primary: '#b44756',
    background: 'rgb(242, 242, 242)',
    cardBackground: 'rgb(255, 255, 255)',
    card: '#b44756',
    text: '#555',
    border: '#b44756',
    notification: 'rgb(255, 59, 48)',
  }
}

export const dark : Theme = {
  device: 'desktop',
  dark: true,
  font: {
    main: 'Roboto-Regular'
  },
  fontSize: {
    searchInput: '16px',
    text: '14px',
    label: '15px',
    inputText: '19px',
    buttonText: '15px'
  },
  opacity: {
    buttonClick: 0.6
  },
  colors: {
    tabActive: '#fff',
    placeholder: 'rgba(255, 255, 255, 0.5)',
    statusBar: '#b44756',
    navigationBar: '#161616',
    notificationTextColor: '#fff',
    error: 'rgb(224, 36, 94)',
    buttonTextColor: '#fff',
    inputText: '#fff',
    inputBackground: 'rgba(255, 255, 255, 0.05)',
    inputBorder: 'rgba(255, 255, 255, 0.08)',
    label: 'rgba(255, 255, 255, 0.5)',
    primary: '#b44756',
    background: '#1c1c1c',
    cardBackground: '#161616',
    card: '#b44756',
    text: '#cccccc',
    border: '#b44756',
    notification: 'rgb(255, 69, 58)',
  }
}

export default function AppThemeProvider({ children }) {
  const isDesktop = useMediaQuery({ minWidth: 920 })
  const themeName = useSystemTheme()

  const theme : any = useMemo(() => {
    return {
      ...(themeName === 'dark' ? dark : light),
      device: isDesktop ? 'desktop' : 'mobile'
    }
  }, [themeName, isDesktop])

  useSetNavBarColor(theme.colors.navigationBar)

  return (
    <ThemeProvider theme={theme}>
      <GlobalStyles theme={theme} />
      <StatusBar
        barStyle='light-content'
        backgroundColor={theme.colors.statusBar} />
      {children}
    </ThemeProvider>
  )
}