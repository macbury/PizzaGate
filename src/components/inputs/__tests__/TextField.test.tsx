import React from 'react'
import { shallow } from 'enzyme'
import snapshot from 'enzyme-to-json'
import { light } from '../../../config/theme'
import TextField, { ITextFieldProps } from '../TextField'

it('renders correctly', () => {
  const props : ITextFieldProps = {
    value: 'test',
    onChangeText: jest.fn(),
    theme: light
  }
  const component = shallow((
    <TextField {...props} />
  ))
  expect(snapshot(component)).toMatchSnapshot()
})