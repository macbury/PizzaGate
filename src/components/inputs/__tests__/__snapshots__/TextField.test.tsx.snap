// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<TextField__Wrapper>
  <TextField__TextInputContainer
    focused={false}
    hasError={false}
  >
    <TextField__Label
      focused={false}
      hasError={false}
    />
    <TextField__TextInput
      onBlur={[Function]}
      onChangeText={[MockFunction]}
      onFocus={[Function]}
      theme={
        Object {
          "colors": Object {
            "activeHeader": "rgba(51, 153, 102, 0.1)",
            "background": "rgb(242, 242, 242)",
            "border": "rgb(216, 216, 216)",
            "buttonTextColor": "#fff",
            "card": "rgb(255, 255, 255)",
            "error": "rgb(224, 36, 94)",
            "inputBackground": "rgba(0, 0, 0, 0.05)",
            "inputBorder": "rgba(0, 0, 0, 0.08)",
            "inputText": "#333",
            "label": "rgba(0, 0, 0, 0.5)",
            "navigationBar": "#000000",
            "notification": "rgb(255, 59, 48)",
            "notificationTextColor": "#fff",
            "placeholder": "rgba(0, 0, 0, 0.3)",
            "primary": "#ff3434",
            "statusBar": "#f2f2f2",
            "text": "#555",
          },
          "dark": false,
          "device": "desktop",
          "font": Object {
            "main": "Roboto-Regular",
          },
          "fontSize": Object {
            "buttonText": "15px",
            "inputText": "19px",
            "label": "15px",
            "searchInput": "16px",
            "text": "16px",
          },
          "opacity": Object {
            "buttonClick": 0.6,
          },
        }
      }
      value="test"
    />
  </TextField__TextInputContainer>
</TextField__Wrapper>
`;
